### PR TITLE
Exceptions/factory

### PR DESCRIPTION
**Pull request by ChrisAusema**
The factory and defined exceptions.

### CODE REVIEW

File: Hestia/app/src/main/java/hestia/backend/exceptions/ExceptionFactory.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.json.JSONObject;
+
+import hestia.backend.exceptions.ServerExceptions.DatabaseException;
+import hestia.backend.exceptions.ServerExceptions.InvalidStateException;
+import hestia.backend.exceptions.ServerExceptions.NotFoundException;
+import hestia.backend.exceptions.ServerExceptions.ServerException;
+import hestia.backend.exceptions.ServerExceptions.SetupFailedException;
+import hestia.backend.exceptions.ServerExceptions.UndefinedServerException;
+
+public class ExceptionFactory {
```
Comment (StefanEvanghelides) [inline Hestia/app/src/main/java/hestia/backend/exceptions/ExceptionFactory.java:15]: Add tests for this class. Note: you should make a test for every case of the switch.

File: Hestia/app/src/androidTest/java/com/rugged/application/hestia/backend/exceptions/ServerExceptions/DataBaseExceptionTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import android.support.test.runner.AndroidJUnit4;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
```
Comment (StefanEvanghelides) [inline Hestia/app/src/androidTest/java/com/rugged/application/hestia/backend/exceptions/ServerExceptions/DataBaseExceptionTest.java:12]: Android Studio creates redundant white spaces in the imports. These are really ugly and unnecessary, in my opinion. I believe we should remove them. What do others think?

File: Hestia/app/src/main/java/hestia/backend/exceptions/ExceptionFactory.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        this.exception=exception;
+    }
+
+
+    public ServerException getException(){
+        String exceptionName=exception.get("exception").getAsString();
+        JsonObject details=exception.get("details").getAsJsonObject();
+        switch(exceptionName){
+            case "DatabaseException":
+                return new DatabaseException(details.get("type").getAsString(),details.get("message").getAsString());
```
Comment (LarsHoldijk) [inline Hestia/app/src/main/java/hestia/backend/exceptions/ExceptionFactory.java:28]: Better to get the information from the json object in different lines to make it easier to read.

File: Hestia/app/src/main/java/hestia/backend/exceptions/ExceptionFactory.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+public class ExceptionFactory {
+    JsonObject exception;
+
+    public ExceptionFactory(JsonObject exception){
+        this.exception=exception;
+    }
+
+
+    public ServerException getException(){
```
Comment (LarsHoldijk) [inline Hestia/app/src/main/java/hestia/backend/exceptions/ExceptionFactory.java:23]: Can be a static method that is given the exception

File: Hestia/app/src/main/java/hestia/backend/exceptions/ServerExceptions/UndefinedServerException.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package hestia.backend.exceptions.ServerExceptions;
+
+public class UndefinedServerException extends ServerException {
+    String message;
+        public UndefinedServerException(){
+            message= "InternalServerError";
+        }
+
+    @Override
+    public String getMessage() {
```
Comment (LarsHoldijk) [inline Hestia/app/src/main/java/hestia/backend/exceptions/ServerExceptions/UndefinedServerException.java:10]: Overriding the method might be a cause for strange behaviour. Better to change the name of the field.

**Review by LarsHoldijk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**